main-option-2.py

```diff
@@ -0,0 +1,171 @@
+
+#Re-run the app to play again
+def restartApp():
+    '''This function will restart the app if the user requests at the end of the previous game'''
+    call(["python", "main-option-2.py"])
+
+#Choose an amount of random outputs
+def addNum():
+    '''This function converts the string input into an int and uses it to select a random number of names from the list'''
+    num_input = input(f"Enter a number between 1 and {len(names)}: ")
+
+    validateNum(num_input)
+        
+    converted = float(num_input[0])
+
+    newN = int(converted)
+
+    while newN > len(names) or newN <= 0:
+
+        num_input = input(f"Number entered must be between 1 and {len(names)} inclusive: ")
+        
+        newN = int(num_input)
+
+    indices = []
+
+    for i in range(newN):
+
+        def newNum():
+
+            rNum = random.randint(1, len(names))
+
+            while rNum in indices:
+
+                rNum = random.randint(1, len(names))
+
+            else:
+
+                return rNum
+            
+        indices.append(newNum())
+
+    output = []
+
+    for j in indices:
+
+        output.append(names[j-1])
+
+    print(output)
+
+    again = input('Would you like to play again? Enter Y for yes or anything else to exit: ')
+
+    if again == 'Y' or again == 'y':
+        restartApp()
+    else:
+        sys.exit()  
+     
+    #return output
+
+#Add names to the list
+def enterName():
+    '''This function allows user to keep adding names to the list after they've been validated'''
+    name_input = input("Enter a name and press enter to add it to the list otherwise just press enter to stop: ")
+
+    if name_input != '':
+
+        validateName(name_input)
+
+    else:
+
+        validateEntKey(name_input)
+
+#Validate reason for pressing enter
+def validateEntKey(name_input):
+    '''This function will validate whether enter key was pressed to add another name or stop adding names'''
+    if name_input == '' and len(names) <= 1:
+
+        print('The list must have two or more names to continue')
+
+        name_input = None
+
+        enterName()
+
+    elif name_input == '' and len(names) > 1:
+
+        name_input = None
+
+        print('Continue')
+
+        addNum()
+
+#Validate users input
+def validateName(name_input):
+    '''This function will validate whether the name entered is made of letters only and is already in the list'''
+    if name_input.isalpha() is not True:
+
+        print('Names entered must contain letters only')
+
+        name_input = None
+
+        enterName()
+
+    elif name_input in names:
+
+        print('No duplicate names are allowed')
+
+        name_input = None
+
+        enterName()
+         
+    else:
+
+        addName(name_input)
+
+        name_input = None
+
+        #print(names) 
+
+#Validate the number entered
+def validateNum(num_input):
+    '''This function will validate the number entered is a number otherwise request a new entry'''
+    if num_input.isdigit() is not True:
```

Comment:
`isdigit()` will return True if there are digits & False otherwise, there is no need to check if it is not True.
You could say, `if not num_input.isdigit():`.